src/libmongoc/tests/test-mongoc-ts-pool.c

```diff
@@ -0,0 +1,97 @@
+#include "mongoc/mongoc-ts-pool-private.h"
+
+#include "TestSuite.h"
+#include "test-libmongoc.h"
+
+static void
+test_ts_pool_empty (void)
+{
+   mongoc_ts_pool *pool = mongoc_ts_pool_new (
+      (mongoc_ts_pool_params){.element_size = sizeof (int)});
+   BSON_ASSERT (mongoc_ts_pool_is_empty (pool));
+   mongoc_ts_pool_free (pool);
+}
+
+
+static void
+test_ts_pool_simple (void)
+{
+   mongoc_ts_pool *pool = mongoc_ts_pool_new (
+      (mongoc_ts_pool_params){.element_size = sizeof (int)});
+   int *item;
+   int *item2;
+
+   item = mongoc_ts_pool_get_existing (pool);
+   BSON_ASSERT (!item);
+
+   item = mongoc_ts_pool_get (pool, NULL);
+   BSON_ASSERT (item);
+   ASSERT_CMPINT (*item, ==, 0);
+   *item = 42;
+   ASSERT_CMPSIZE_T (mongoc_ts_pool_size (pool), ==, 0);
+   mongoc_ts_pool_return (item);
+   ASSERT_CMPSIZE_T (mongoc_ts_pool_size (pool), ==, 1);
+
+   item2 = mongoc_ts_pool_get_existing (pool);
+   BSON_ASSERT (item2);
+   ASSERT_CMPINT (*item2, ==, 42);
+   ASSERT_CMPSIZE_T (mongoc_ts_pool_size (pool), ==, 0);
+
+   mongoc_ts_pool_drop (item2);
+   ASSERT_CMPSIZE_T (mongoc_ts_pool_size (pool), ==, 0);
+
+   mongoc_ts_pool_free (pool);
+}
+
+static int
+_is_int_42 (int *v, void *unused)
+{
+   (void) unused;
+   return *v == 42;
+}
```

Comment:
```suggestion
static int
_is_int_42 (int *v, BSON_MAYBE_UNUSED void *unused)
{
   return *v == 42;
}
```
Is silencing `-Wunused-parameter` warnings even necessary given there are currently +600 instances of said warning throughout the C driver? (I suppose no time like the present to start reducing them.)

Reply:
I have a stashed changeset that marks every single unused parameter in the codebase. The "maybe unused" attribute doesn't help on MSVC, but my stashed changes will fix for MSVC as well. A good thing for a new PR, but I don't want to pile on _another_ PR on top of my four open ones already 🙂.

Reply:
Fair enough! Will consider further modifications regarding unused warnings optional.